docs/specification.md

```diff
@@ -357,6 +463,158 @@ unit
 
 ## Proposal entrypoints
 
-**TODO**
+### **propose**
+
+```haskell
+data Proposal proposalMetadata = Proposal
+    { frozen_token_requirement :: Natural
+    -- ^ How many tokens should be frozen in order to vote on this proposal.
+    , metadata :: proposalMetadata
+    }
+
+data Parameter proposalMetadata
+  = Propose Natural (Proposal proposalMetadata)
+```
+
+Parameter (in Michelson):
+```
+(pair %propose
+  nat
+  (pair
+    (nat %frozen_token_requirement)
+    (<proposal_type> %metadata)
+  )
+)
+```
+
+- The proposal is saved and assigned a unique numeric proposal ID.
+- The `Natural` value (let's say `n`) determines how many sender's tokens will be frozen.
```

Comment:
Should this `Natural` (`proposal_token_requirement`) be a part of the contract runtime configuration instead of the proposal?

I think it's something that the DAO owner would want to specify.

Reply:
My idea is that the DAO owner configures `cProposalCheck` lambda that determines how many tokens are required. In simplest case it may require a specific contracts amount of tokens to be frozen and fail if the amount is different. However, I can imagine that it can be more complex in some cases, e. g. it may depend on the actual proposal. So the person can set arbitrary `frozen_token_requirement`, but this value must pass `cProposalCheck`.

Reply:
I see, that makes sense. Thanks for the explanation, Ivan.